pkgs/applications/emulators/c64-debugger/default.nix

```diff
@@ -0,0 +1,90 @@
+{ fetchgit, lib, pkgs, stdenv }:
+
+stdenv.mkDerivation {
+  name = "c64-debugger";
+  version = "0.64.58.6";
+
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/c64-debugger/code";
+    rev = "f97772e3f5c8b4fa99e8ed212ed1c4cb1e2389f1";
+    sha256 = "sha256-3SR73AHQlYSEYpJLtQ/aJ1UITZGq7aA9tQKxBsn/yuc=";
+  };
+
+  buildInputs = [
+    pkgs.alsa-lib
```

Comment:
Out of curiousity, why are you using `pkgs.`?

This is not the first time I see a novice using this hack...

Reply:
I guess it is a habit that comes from NixOS configuration files. Why is it bad and should not be used?

Reply:
Hard to explain without using buzzwords.

First, finer control. Passing the whole `pkgs` doesn't aldow to pick different

Reply:
Hard to explain without buzzwords.

- finer control of inputs.

Passing the whole `pkgs` as a global attribute hinders the possibility of choosing each input. (This can be circumvented, tho.)

- cross-compilation.

  Sometimes the packages used as `buildInputs` are distinct from those used as `nativeBuildInputs`, because the platform used for compilation is not necessarily the same used for running the resulting executable. 

  Therefore the `pkgs` used at build time (usually listed under `nativeBuildInputs`) are different from the `pkgs` used at runtime (usually `buildInputs`). 

  So, passing `pkgs` as a global attribute forces both to be the same, making the derivation less amenable to cross compilation.

Reply:
Makes sense, thanks for the explanation.